lib/langchain/dependency_helper.rb

```diff
@@ -26,8 +28,13 @@ def depends_on(gem_name)
         raise VersionError, "The #{gem_name} gem is installed, but version #{gem_requirement} is required. You have #{gem_version}."
       end
 
+      lib_name = gem_name if req == true
+      lib_name = req if req.is_a?(String)
```

Comment:
Added the explict `req == true` check so the order doesn't affect the result. Otherwise, if we had only:

```rb
lib_name = gem_name if req
lib_name = req if req.is_a?(String)
```

... it would also work, but order would matter.